File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i in range(len(nums)):
            key = target - nums[i]
            if key in hashmap:
                return [hashmap[key], i]
            else:
                hashmap[nums[i]] = i
                
                

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexMap = new HashMap<>();
        for(int i=0; i < nums.length; i++) {
            Integer key = (Integer)target - nums[i];
            if(indexMap.containsKey(key)) {
                return new int[] {indexMap.get(key), i};
            }
            indexMap.put((Integer)nums[i], i);
        }
        return null;
    }
}